internal/cmd/kafka/command_topic_produce.go

```diff
@@ -127,7 +132,11 @@ func (c *command) produce(cmd *cobra.Command, args []string) error {
 	signal.Notify(signals, os.Interrupt)
 	go func() {
 		<-signals
-		close(input)
+		select {
+		case <-input:
+		default:
+			close(input)
```

Comment:
Is it still possible to have a double close? I thought the solution would involve removing one of the `close()` functions.

Reply:
It might be possible with very precise timing.

And both of these `close()` calls appear to be necessary without a larger refactor: without closing the input in the signal goroutine, the scanner goroutine treats `Ctrl-C` like a regular empty line on Unix, which means that it doesn't exit (on Windows, the scanner apparently treats `Ctrl-C` like EOF so it would still end there).

Without closing the input in the scanner goroutine, `Ctrl-D` no longer exits on Unix since it isn't an interrupt signal.

The easiest solution for now is to just repeat this check for all `close` calls.

Reply:
Nevermind; that idea simply introduces its own possibilities of double closure panic.

At this point I'm just going to resell this as fixing a panic that occurs in `produce` for Windows (since it happens nearly 100% of the time as far as I can tell) and leave the CLI-2642 ticket open.